sway-core/src/type_system/info.rs

```diff
@@ -500,6 +508,9 @@ impl OrdWithEngines for TypeInfo {
             ) => l_trait_type_id
                 .cmp(r_trait_type_id)
                 .then_with(|| l_name.cmp(r_name)),
+            (Self::Ref(l_ty), Self::Ref(r_ty)) => type_engine
```

Comment:
Same thing here... others branches are ordering the type id and not the `TypeInfo` struct.

Reply:
Hmmm, here I am not that sure. Some branches like `TraitType` do compare `TypeId`s but some other like `Alias` or `Array` compare `TypeInfo`s ;-) IMO it makes sense to compare `TypeId`s everywhere but I want to give it a closer look before changing it.